.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,46 @@
+name: Build and push Docker image
+
+on:
+  push:
+    branches:
```

Comment:
We want to still build this image during PRs, but only push when its been merged into master. That way we know for sure it'll successfully push once merged

Reply:
@HenryNguyen5 due to limitations Github Actions syntax, I decided to split into two workflows, one that only builds PRs and other that  builds and publishes branches and tags.